### PR TITLE
[huggingface][pytorch][training][fix] Patch integ tests for hf pt 2.0 training image

### PR DESCRIPTION
**Pull request by jinyoung-lim**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [x] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

### Formatting
- [x] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ec2_benchmark_tests = true` or `sagemaker_benchmark_tests = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jinyoung-lim** [comment]
SageMaker test successful: https://tiny.amazon.com/3zp96vis/IsenLink

**Comment by jinyoung-lim** [comment]
New Pillow vuln needs to be addressed: https://tiny.amazon.com/rptjtzeu/IsenLink

**Comment by jinyoung-lim** [comment]
local test: https://tiny.amazon.com/s25y45ov/IsenLink
sagemaker test: https://tiny.amazon.com/15md7w0ov/IsenLink
sanity test: https://tiny.amazon.com/15nxrh2p9/IsenLink

**Comment by jinyoung-lim** [comment]
Asking about local test failure with test_smmp: https://huggingface.slack.com/archives/C01GJG9PE12/p1706738511902109

**Comment by jinyoung-lim** [comment]
local: https://tiny.amazon.com/18nx9c9rm/IsenLink
sm: https://tiny.amazon.com/moljm9nl/IsenLink
sanity: https://tiny.amazon.com/1j1xfgadh/IsenLink

**Comment by jinyoung-lim** [comment]
local test: https://tiny.amazon.com/t60d4yi6/IsenLink
sagemaker: https://tiny.amazon.com/1jpck1qj3/IsenLink
sanity: https://tiny.amazon.com/pnrcmfcd/IsenLink

**Comment by jinyoung-lim** [comment]
Sanity test failure with:
```
self = <[AttributeError("'SpecifierSet' object has no attribute '_prereleases'") raised in repr()] SpecifierSet object at 0x7f0e4fb8a730>



specifiers = ['<1.26.17'], prereleases = None
```

**Comment by jinyoung-lim** [comment]
sagemaker test for test_smmp_* failed with:
```
<html><head></head><body>
[1,mpirank:0,algo-1]<stderr>:TensorSplitError: Batch size must be divisible by the number of microbatches.
--


[1,mpirank:0,algo-1]&lt;stderr&gt;:<span class="ace_link">https://sagemaker.readthedocs.io/en/stable/api/training/smd_model_parallel.html</span></body></html>
```

**Comment by jinyoung-lim** [comment]
sagemaker test SUCCESS! https://tiny.amazon.com/y0hk9u61/IsenLink

File: test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smmp.py [hunk 1]
```diff
@@ -26,8 +26,8 @@
 hyperparameters = {
     "model_name_or_path": "hf-internal-testing/tiny-random-RobertaModel",
     "task_name": "mnli",
-    "per_device_train_batch_size": 1,
-    "per_device_eval_batch_size": 1,
+    "per_device_train_batch_size": 2,  # batch sizes must be divisible by microbatche size
```
Comment (mabunday) [inline test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smmp.py:29]: Not fully following this - what isn't divisible by 1?

Comment (jinyoung-lim) [reply test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smmp.py:29]: @mabunday  Without this change, we get below error. Note that the microbatch size is set to 2. And `2 % 1 = 1 != 0`. 


```
[1,mpirank:2,algo-1]<stderr>:TensorSplitError: Batch size must be divisible by the number of microbatches. 
[1,mpirank:2,algo-1]<stderr>:Found batch size=1, number of microbatches=2. If this is the last batch of the 
```

Comment (jinyoung-lim) [reply test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smmp.py:29]: https://github.com/aws/deep-learning-containers/blob/e1adf2c467c7bd1faf43fc867de233b91fc947d5/test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smmp.py#L47

Comment (mabunday) [reply test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smmp.py:29]: nit: Can we define a variable and set it so they're the same (or derive one from the other) then if there's this implicit dependency?

Comment (jinyoung-lim) [reply test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smmp.py:29]: Hmm sorry not sure if I get your point - could you elaborate?

**Review by amzn-choeric — DISMISSED** [review]

**Review by amzn-choeric — APPROVED** [review]

**Pull request merged**